### DISCUSSION TITLE
Compile-time error when mapping nullable to non-nullable

### DISCUSSION THREAD

**Discussion by fernetowac** [riok/mapperly#375] [Ideas]
When I map nullable property to non-nullable one, I'd prefer to get a compile-time error. I'd like this to be the default behavior unless explicitly stated else.

**Comment by latonz** [comment]
This would definitely be a cool feature of mapperly. Kinda relates to https://github.com/riok/mapperly/issues/98.

**Reply by latonz** [reply]
Created https://github.com/riok/mapperly/issues/602

**Comment by R-abodyak** [comment]
the default behavior is mapping null to default value 
however you can explicitly change that :
 [Mapper(AllowNullPropertyAssignment  = false,
    ThrowOnPropertyMappingNullMismatch = true)]

for more info refer to https://mapperly.riok.app/docs/configuration/mapper/#null-values
